File: app/src/main/java/ua/com/anyapps/kt_pirates_notes/di/component/ApplicationComponent.java
package ua.com.anyapps.kt_pirates_notes.di.component;

import javax.inject.Singleton;

import dagger.Component;
import ua.com.anyapps.daggerprovidesharedpreferences.MainActivity;
import ua.com.anyapps.daggerprovidesharedpreferences.di.module.ApplicationModule;
import ua.com.anyapps.daggerprovidesharedpreferences.di.module.TestModule;

@Singleton
@Component(modules = {ApplicationModule.class, TestModule.class})
public interface ApplicationComponent {
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/ua/com/anyapps/kt_pirates_notes/di/component/ApplicationComponent.kt
package ua.com.anyapps.kt_pirates_notes.di.component

import dagger.Component
import ua.com.anyapps.kt_pirates_notes.di.module.AppModule
import ua.com.anyapps.kt_pirates_notes.di.module.RoomModule
import ua.com.anyapps.kt_pirates_notes.repository.RepositoryImpl
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
    AppModule::class, RoomModule::class))
interface ApplicationComponent {
    fun inject(into: RepositoryImpl)
}